akka-actor-tests/src/test/scala/akka/io/TcpIntegrationSpecSupport.scala

```diff
@@ -11,10 +11,14 @@ import akka.actor.ActorRef
 import akka.io.Inet.SocketOption
 import akka.testkit.SocketUtil._
 import Tcp._
+import akka.actor.ActorSystem
 
 trait TcpIntegrationSpecSupport { _: AkkaSpec ⇒
 
-  class TestSetup(shouldBindServer: Boolean = true) {
+  class TestSetup(shouldBindServer: Boolean = true, runClientInExtraSystem: Boolean = true) {
+    val clientSystem =
```

Comment:
Will need shutdown or will leak for sbt shell

Reply:
Thanks, good point, will fix.

Reply:
Now the clientSystem is terminated when the main system used for the server is terminated as well.